forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,525 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+
+/// A controller that stores the expanded state of an [_Item].
+interface class FAccordionController extends ChangeNotifier {
+  /// The duration of the expansion and collapse animations.
+  final Duration duration;
+  final List<(AnimationController, Animation)> _controllers;
+  final Set<int> _values;
+  final int _min;
+  final int? _max;
+
+  /// Creates a [FAccordionController].
+  ///
+  /// The [min] and [max] values are the minimum and maximum number of selections allowed. Defaults to no minimum or maximum.
+  ///
+  /// # Contract:
+  /// * Throws [AssertionError] if [min] < 0.
+  /// * Throws [AssertionError] if [max] < 0.
+  /// * Throws [AssertionError] if [min] > [max].
+  FAccordionController({
+    int min = 0,
+    int? max,
+    Set<int>? values,
```

Comment:
```suggestion
    Set<int> values = {},
```

Reply:
Should we rename it to `initialIndexes`?

Reply:
What if we provide `initialIndexes` that is outside the range of min and max?

Reply:
The reason I made it nullable is because  `Set<int> values = {}` requires the set to be prefixed with `const` but this makes it unmodifiable later